base/reshapedarray.jl

```diff
@@ -292,3 +292,16 @@ substrides(strds::NTuple{N,Int}, I::Tuple{ReshapedUnitRange, Vararg{Any}}) where
     (size_to_strides(strds[1], size(I[1])...)..., substrides(tail(strds), tail(I))...)
 unsafe_convert(::Type{Ptr{T}}, V::SubArray{T,N,P,<:Tuple{Vararg{Union{RangeIndex,ReshapedUnitRange}}}}) where {T,N,P} =
     unsafe_convert(Ptr{T}, V.parent) + (first_index(V)-1)*sizeof(T)
+
+
+_checkcontiguous(::Type{Bool}, A::AbstractArray) = size_to_strides(1, size(A)...) == strides(A)
```

Comment:
Will this break for
```julia
struct ReversedUInt8Vec3 <: DenseVector{UInt8}
    a::UInt8
    b::UInt8
    c::UInt8
end
Base.getindex(v::ReversedUInt8Vec3, i::Int) = i == 1 ? v.c :
    i == 2 ? v.b :
    i == 3 ? v.a : throw_boundserror(v, i)
Base.strides(v::ReversedUInt8Vec3) = (-1,)
```

Reply:
`reshape(v::ReversedUInt8Vec3, sz)` should be handled via L304.
Thus `_checkcontiguous` is not used.

I thought `stride(a::DenseArray, 1) === 1`, but it might not be true.
So L299 should be restricted to `_checkcontiguous(::Type{Bool}, A::Array) = true`